packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -799,8 +799,9 @@ export const MenuBarMixin = (superClass) =>
       e.stopPropagation();
       const button = this._getButtonFromEvent(e);
       if (button) {
-        this.__openSubMenu(button, false);
+        this.__openSubMenu(button, button.__triggeredWithActiveKeys);
       }
+      button.__triggeredWithActiveKeys = null;
```

Comment:
Would be cleaner to clear this in the button itself after the super call.

Reply:
Moved it to he button.